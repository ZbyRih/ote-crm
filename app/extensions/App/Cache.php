<?php
namespace App\Extensions\App;

use Nette\Caching\IStorage;

class Cache extends \Nette\Caching\Cache{

	public function __construct(IStorage $storage, $namespace = null){
		parent::__construct($storage, 'app-runtime-cache');
	}
}