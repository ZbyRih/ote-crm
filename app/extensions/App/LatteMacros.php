<?php

namespace App\Extensions\App;

use Latte\Compiler;
use Latte\MacroNode;
use Latte\PhpWriter;
use Latte\Macros\MacroSet;
use Latte\MacroTokens;

class LatteMacros extends MacroSet{

	public static function install(
		Compiler $compiler)
	{
		$set = new static($compiler);

		$set->addMacro('portlet', [
			$set,
			'portletBegin'
		], [
			$set,
			'portletEnd'
		]);

		$set->addMacro('portlet_break', [
			$set,
			'portletBreak'
		]);

		$set->addMacro('portlet_end', [
			$set,
			'portletEnd'
		]);

		$set->addMacro('caption_title', [
			$set,
			'captionTitle'
		]);

		$set->addMacro('flashes', [
			$set,
			'flashes'
		]);

		$set->addMacro('wpm_script', [
			$set,
			'wpmScript'
		]);

		$set->addMacro('wpm_image', [
			$set,
			'wpmImage'
		]);

		$set->addMacro('wpm_imageBase64', [
			$set,
			'wpmImageBase64'
		]);

		$set->addMacro('wpm_file', [
			$set,
			'wpmFile'
		]);

		return $set;
	}

	public function portletBreak(
		MacroNode $node,
		PhpWriter $writer)
	{
		$node->openingCode = '</div></div>';
		$this->portletBegin($node, $writer);
	}

	public function portletBegin(
		MacroNode $node,
		PhpWriter $writer)
	{
		$class = '';
		if(!empty($node->args)){
			$rest = '';
			$t = new MacroTokens($node->args);
// 			$t->nextUntil('class');
			$rest = $t->joinUntil('class');
			if($t->isNext() && $t->nextValue() == 'class'){
				$t->nextUntil(MacroTokens::T_STRING);
				$class = trim($t->expectNextValue(), '\'"');
				$rest .= $t->joinAll();
				$node->args = $rest;
			}
		}

		$node->openingCode .= '<div class="portlet light bordered ' . $class . '">' . PHP_EOL;

		if(!empty($node->args)){

			$node->openingCode .= '<div class="portlet-title">';
			$this->captionTitle($node, $writer);
			$node->openingCode .= '</div>';
		}

// 		$node->openingCode .= '<div class="portlet-body form clearfix">' . PHP_EOL;
		$node->openingCode .= '<div class="portlet-body clearfix">' . PHP_EOL;
	}

	public function portletEnd(
		MacroNode $node,
		PhpWriter $writer)
	{
		$node->closingCode = '</div></div>';
	}

	public function captionTitle(
		MacroNode $node,
		PhpWriter $writer)
	{
		$node->openingCode .= '<div class="caption">' . PHP_EOL . '	<span class="caption-subject font-green bold uppercase">' . PHP_EOL;
		if($node->modifiers){
			$node->openingCode .= '<?php ' . $writer->write('echo %modify(%node.args);') . '?>';
		}else{
			$node->openingCode .= '<?php ' . $writer->write('echo %escape(%node.args);') . '?>';
		}
		$node->openingCode .= '</span>' . PHP_EOL . '</div>';
	}

	public function flashes(
		MacroNode $node,
		PhpWriter $writer)
	{
		$node->openingCode .= '<?php $flashesUuid = \Nette\Utils\Random::generate(10); ?>' . PHP_EOL;
		$node->openingCode .= '<div class="row" id="<?php echo $flashesUuid ?>">' . PHP_EOL;
		$node->openingCode .= '	<div class="col-md-12 flashes">' . PHP_EOL;
		$node->openingCode .= '		<script>' . PHP_EOL;
		$node->openingCode .= '		(function(){
										var p = document.getElementById(\'<?php echo $flashesUuid ?>\');
										p.style.display = \'none\';
										p.style.visibility = \'hidden\';
									}).call();' . PHP_EOL;
		$node->openingCode .= '		</script>' . PHP_EOL;
		$node->openingCode .= '<?php ' . $writer->write('foreach(%node.word as $f){') . ' ?>' . PHP_EOL;
		$node->openingCode .= '		<div n:foreach="$flashes as $flash" class="alert alert-<?php ' . $writer->write('echo $f->type;') . '?>">' . PHP_EOL;
		$node->openingCode .= '<?php echo ' . $writer->write('$f->message;') . ' ?>' . PHP_EOL;
		$node->openingCode .= '		</div>' . PHP_EOL;
		$node->openingCode .= '<?php ' . $writer->write('}') . '?></div></div>' . PHP_EOL;
	}

	public function wpmScript(
		MacroNode $node,
		PhpWriter $writer)
	{
		return $writer->write('echo \App\Extensions\App\WebPackManifest::script(%node.word, %node.array);');
	}

	public function wpmImage(
		MacroNode $node,
		PhpWriter $writer)
	{
		return $writer->write('echo \App\Extensions\App\WebPackManifest::image(%node.word, %node.array);');
	}

	public function wpmImageBase64(
		MacroNode $node,
		PhpWriter $writer)
	{
		return $writer->write('echo \App\Extensions\App\WebPackManifest::imageBase64(%node.word, %node.array);');
	}

	public function wpmFile(
		MacroNode $node,
		PhpWriter $writer)
	{
		return $writer->write('echo \App\Extensions\App\WebPackManifest::file(%node.word, %node.array);');
	}
}