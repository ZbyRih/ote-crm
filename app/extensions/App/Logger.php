<?php

namespace App\Extensions\App;

use greeny\NetteSlackLogger\SlackLogger;
use greeny\NetteSlackLogger\MessageFactory;
use Nette\Http\Url;
use Nette\Utils\Strings;
use Nette\DI\Container;
use Tracy\Debugger;

class Logger extends \Tracy\Logger{

	const STAT = 'stat';

	private $defaults = [
		'enabled' => FALSE,
		'timeout' => 30,
		'messageFactory' => MessageFactory::class,
		'defaults' => [
			'channel' => NULL,
			'icon' => NULL,
			'name' => NULL,
			'title' => NULL,
			'text' => NULL,
			'color' => NULL
		]
	];

	/** @var SlackLogger */
	private $slackLogger;

	public function __construct(Container $container){
		parent::__construct(Debugger::$logDirectory, Debugger::$email, Debugger::getBlueScreen());
		$this->createSlackLogger($container);
	}

	private function createSlackLogger(Container $container){
		$config = $container->getParameters();

		if(!isset($config['slackLogger']) || !isset($config['slackLogger']['enable']) || !$config['slackLogger']['enable']){
			return;
		}

		$config = $config['slackLogger'];

		$config = \Nette\DI\Config\Helpers::merge($config, $this->defaults);

		if($config['defaults']['name'] && Strings::startsWith($config['defaults']['name'], 'http')){
			$config['defaults']['name'] = (new Url($container->parameters['app']['basePath']))->host;
		}

		$this->slackLogger = new SlackLogger($config['slackUrl'], new MessageFactory($config['defaults'], $config['logUrl']), $config['timeout']);
	}

	public function log($value, $priority = self::INFO){
		if($priority == self::STAT){
			// poslat jen na ELK
			// 			parent::log(json_encode($value), $priority);
		}else{
			if($this->slackLogger){
				$this->slackLogger->log($value, $priority);
			}
			parent::log($value, $priority);
		}
	}

	public function getSlack(){
		return $this->slackLogger;
	}

	public function getSection($namespace){
		return (new LoggerSection($this))->setNameSpace($namespace);
	}
}