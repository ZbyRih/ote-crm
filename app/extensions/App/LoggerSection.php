<?php
namespace App\Extensions\App;

use Tracy\ILogger;
use Nette\Utils\Strings;

class LoggerSection{

	/** @var ILogger */
	private $logger;

	/** @var string */
	private $namespace;

	/**
	 *
	 * @param ILogger $logger
	 */
	public function __construct(ILogger $logger){
		$this->logger = $logger;
	}

	/**
	 *
	 * @param string $namespace
	 * @return self
	 */
	public function setNameSpace($namespace){
		$this->namespace = $namespace;
		return $this;
	}

	public function log($value){
		if(is_string($value)){
			$value = trim($value, '"');
			if(!Strings::startsWith($value, 'FALSE') && !Strings::startsWith($value, 'TRUE') && !Strings::startsWith($value, 'ERROR')){
				$value = '"' . $value;
			}
			if(Strings::contains($value, '"')){
				$value .= '"';
			}
		}else if(is_array($value)){
			reset($value);
			$first = current($value);
			if($first === 'FALSE' || $first === 'TRUE' || $first === 'ERROR'){
				array_shift($value);
				$value = $first . ' "' . trim(implode('" "', $value), '"') . '"';
			}else{
				$value = '"' . trim(implode('" "', $value), '"') . '"';
			}
		}

		$this->logger->log($value, $this->namespace);
	}
}