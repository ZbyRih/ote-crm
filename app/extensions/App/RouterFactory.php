<?php

namespace App\Extensions\App;

use Nette;
use Nette\Application\Routers\RouteList;
use Nette\Application\Routers\Route;

class RouterFactory{

	/**
	 *
	 * @return Nette\Application\IRouter
	 */
	public static function createRouter(){
		$router = new RouteList();
		$router[] = new Route('admin/index.php', [
			'presenter' => 'Legacy',
			'action' => 'default',
			'do' => null
		]);

		$router[] = new Route('admin[/]', [
			'presenter' => 'Legacy',
			'action' => 'default',
			'do' => null
		]);

// 		$router[] = new Route('admin[/]',
// 			[
// 				'presenter' => 'Legacy',
// 				'action' => 'default',
// 				null => [
// 					Route::FILTER_IN => [
// 						'\App\Extensions\App\RouterFactory',
// 						'inFilterLegacy'
// 					]
// 				]
// 			], Route::ONE_WAY);

// 		$router[] = new Route('legacy/',
// 			[
// 				'presenter' => 'Legacy',
// 				'action' => 'default',
// 				null => [
// 					Route::FILTER_IN => [
// 						'\App\Extensions\App\RouterFactory',
// 						'inFilterLegacy'
// 					]
// 				]
// 			], Route::ONE_WAY);

		$router[] = new Route('<presenter>/<action>[/<id>]', 'Homepage:default');
		$router[] = new Route('<module>/<presenter>/<action>[/<id>]', ':Homepage:default');
		return $router;
	}
}