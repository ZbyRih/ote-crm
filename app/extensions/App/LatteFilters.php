<?php

namespace App\Extensions\App;

use App\Extensions\Utils\Strings;

class LatteFilters{

	public static function common($filter, $value){
		if(method_exists(__CLASS__, $filter)){
			$args = func_get_args();
			array_shift($args);
			$ret = call_user_func_array([
				__CLASS__,
				$filter
			], $args);
			if($ret === null){
				return '';
			}
			return $ret;
		}
		throw new \Exception('Neexistující filter: \'' . $filter . '\'');
	}

	public static function uno($s, $null){
		return ($s) ? $s : $null;
	}

	public static function bin($s, $a, $b){
		return ($s) ? $a : $b;
	}

	public static function nparam($p, $param, $null = null){
		return ($p) ? $p[$param] : $null;
	}

	public static function nempty($s, $p){
		return !(empty($s)) ? $s : $p;
	}

	public static function nisset($s, $p){
		return !(isset($s)) ? $s : $p;
	}

	public static function price_curr($c, $dec = 2, $space = '&nbsp;'){
		return number_format($c, $dec, ',', $space) . $space . 'Kč';
	}

	public static function price($c, $dec = 2, $space = '&nbsp;'){
		return number_format($c, $dec, ',', $space);
	}

	public static function dformat($c, $format = 'j.n. Y'){
		if($c){
			return $c->format($format);
		}
		return null;
	}

	public static function html_ent($obsah){
		return htmlentities($obsah, ENT_QUOTES, 'UTF-8');
	}

	public static function round($v, $d = 2){
		return round($v, $d);
	}

	public static function hlnondef($v, $def){
		return ($v != $def) ? '<b class="text-primary">' . $v . '</b>' : $v;
	}

	public static function ibytes($kibytes, $precision = 2){
		static $units = array(
			'B',
			'KiB',
			'MiB',
			'GiB',
			'TiB',
			'PiB'
		);
		$unit = '';
		$kibytes = round($kibytes);
		foreach($units as $unit){
			if(abs($kibytes) < 1024 || $unit === end($units)){
				break;
			}
			$kibytes = $kibytes / 1024;
		}
		return round($kibytes, $precision) . ' ' . $unit;
	}

	public static function firstLower($s){
		return Strings::firstLower($s);
	}
}