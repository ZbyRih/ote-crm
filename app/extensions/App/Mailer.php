<?php
namespace App\Extensions\App;

use App\Models\Repositories\ParametersRepository;
use Nette\Mail\IMailer;
use Nette\Mail\SendException;
use Tracy\Debugger;
use Tracy\ILogger;

class Mailer{

	/** @var IMailer */
	private $mailer;

	/** @var LoggerSection */
	private $logger;

	private $mailerFrom;

	private $mailerName;

	function __construct(IMailer $mailer, ParametersRepository $options, ILogger $logger){
		$from = $options->mail['from'];
		$this->mailerFrom = $from['email'];
		$this->mailerName = $from['name'];
		$this->mailer = $mailer;
		$this->logger = $logger->getSection('emails');
	}

	/**
	 *
	 * @param \Nette\Bridges\ApplicationLatte\Template $tpl
	 * @param string $toMail
	 * @param string $toName
	 * @param string $subject
	 */
	public function send($tpl, $toMail, $toName, $subject){
		$log = function ($state, $message = '', $msgId = null) use ($subject, $toMail){
			$this->logger->log([
				$state,
				$message,
				$subject,
				$toMail
			]);
		};

		if(!\Nette\Utils\Validators::isEmail($toMail)){
			$log('FALSE', 'invalid email');
			return false;
		}
		try{

// 			$m = new \App\Extensions\Utils\Helpers\Message();

			$m->addTo(trim($toMail), $toName)
				->setFrom($this->mailerFrom, $this->mailerName)
				->setSubject($subject)
				->setHtmlBody((string) $tpl);

			$this->mailer->send($m);

			$log('TRUE', 'send');
			return true;
		}catch(SendException $e){
			$log('ERROR', 'SendException: ' . $e->getMessage());
			Debugger::log($e, Debugger::ERROR);
		}catch(\Exception $e){
			$log('ERROR', 'Exception: ' . $e->getMessage());
			Debugger::log($e, Debugger::ERROR);
		}
		return false;
	}
}