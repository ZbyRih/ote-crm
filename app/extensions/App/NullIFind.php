<?php

namespace App\Extensions\App;

use App\Extensions\Interfaces\ITableFind;

class NullIFind implements ITableFind{

	public function find($id){
		return null;
	}
}