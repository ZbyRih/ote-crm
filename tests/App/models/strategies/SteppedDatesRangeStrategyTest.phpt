<?php

namespace Tests\App\Models\Strategies;

require_once __DIR__ . '/../../../bootstrap.php';

use App\Models\Enums\DatesRangeStepsEnums;
use App\Models\Strategies\SteppedDatesRangeStrategy;
use Tester\Assert;
use Tester\TestCase;
use Carbon\Carbon;

class SteppedDatesRangeStrategyTest extends TestCase{

	public function __construct()
	{
	}

	public function testCreate()
	{
		$str = new SteppedDatesRangeStrategy();
		$str->setFrom(new \DateTime('1970-01-01'));
		$str->setTo(new \DateTime('1970-12-31'));
		$str->setInterval(DatesRangeStepsEnums::STEP_QUATER);

		$items = $str->generate();

		Assert::same(4, count($items));

		$str = new SteppedDatesRangeStrategy();
		$str->setFrom(new \DateTime('1970-02-01'));
		$str->setTo(new \DateTime('1970-11-30'));
		$str->setInterval(DatesRangeStepsEnums::STEP_QUATER);

		$items = $str->generate();

		Assert::same(4, count($items));
		Assert::equal(new Carbon('1970-02-01'), $items[0]['od']);
		Assert::equal(new Carbon('1970-11-30'), $items[3]['do']);

		$str = new SteppedDatesRangeStrategy();
		$str->setFrom(new \DateTime('1970-04-01'));
		$str->setTo(new \DateTime('1970-11-30'));
		$str->setInterval(DatesRangeStepsEnums::STEP_QUATER);

		$items = $str->generate();

		Assert::same(3, count($items));
		Assert::equal(new Carbon('1970-04-01'), $items[0]['od']);
		Assert::equal(new Carbon('1970-11-30'), $items[2]['do']);

		$str = new SteppedDatesRangeStrategy();
		$str->setFrom(new \DateTime('1970-01-01'));
		$str->setTo(new \DateTime('1970-12-31'));
		$str->setInterval(DatesRangeStepsEnums::STEP_DEFAULT);

		$items = $str->generate();

		Assert::same(12, count($items));

		$str = new SteppedDatesRangeStrategy();
		$str->setFrom(new \DateTime('1970-02-01'));
		$str->setTo(new \DateTime('1970-11-30'));
		$str->setInterval(DatesRangeStepsEnums::STEP_DEFAULT);

		$items = $str->generate();

		Assert::same(10, count($items));
	}
}

(new SteppedDatesRangeStrategyTest())->run();