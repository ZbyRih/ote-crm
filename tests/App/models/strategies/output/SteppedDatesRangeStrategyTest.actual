Carbon\Carbon::__set_state(/* #bb08 */ [
	'date' => '1970-11-30 00:00:00.000000',
	'timezone_type' => 3,
	'timezone' => 'Europe/Prague',
])
