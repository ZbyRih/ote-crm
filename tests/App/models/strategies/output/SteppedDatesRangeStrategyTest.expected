DateTime::__set_state(/* #4786 */ [
	'date' => '1970-11-30 00:00:00.000000',
	'timezone_type' => 3,
	'timezone' => 'Europe/Prague',
])
